packages/protocol/engine/ledger/utxoledger/statediffs.go

```diff
@@ -288,10 +288,14 @@ func (s *StateDiffs) storeTransaction(transaction utxo.Transaction, metadata *me
 	return nil
 }
 
-func (s *StateDiffs) outputWithMetadata(outputID utxo.OutputID) (outputWithMetadata *mempool.OutputWithMetadata) {
+func (s *StateDiffs) outputWithMetadata(outputID utxo.OutputID, index ...slot.Index) (outputWithMetadata *mempool.OutputWithMetadata) {
```

Comment:
Do we need this initializer then? We could just maybe set the index from the outside on the created object. It is just a very specific usecase now: and probably not needed as the outputs metadata should be already set.

Reply:
Done